commit eebbc8945b52dde98445e0f87aeffdf0992e0354
Author: Björn Esser <besser82@fedoraproject.org>
Date:   Sun Jan 14 22:36:53 2018 +0100

    vtkSMSettings: Fix build with jsoncpp-1.8.4
    
    Fixes #17909

diff --git a/ParaViewCore/ServerManager/Core/vtkSMSettings.cxx b/ParaViewCore/ServerManager/Core/vtkSMSettings.cxx
index b6e2b5592a..226180a5ba 100644
--- a/ParaViewCore/ServerManager/Core/vtkSMSettings.cxx
+++ b/ParaViewCore/ServerManager/Core/vtkSMSettings.cxx
@@ -788,7 +788,11 @@ public:
           this->GetSettingBelowPriority(propertySettingCString, highestPriority);
         if (lowerPriorityValue.isNull())
         {
-          if (!proxyValue.removeMember(property->GetXMLName()).isNull())
+          // Can't use reference here, since we need a valid empty object for
+          // deep-copying the removed member.
+          Json::Value removedValue;
+          if (proxyValue.removeMember(property->GetXMLName(), &removedValue) &&
+            !removedValue.isNull())
           {
             this->Modified();
           }
